From e2027ed45f5af46495a6fcef7de86b2a1271c41b Mon Sep 17 00:00:00 2001
From: AndroidMeda <hephappy@gmail.com>
Date: Sun, 6 Jan 2013 23:12:32 +0200
Subject: [PATCH] releasetools: Move packages into extras

Instead of removing packages from makefiles,
move them into extras folder in the ota(zip) file.

By default, build/tools/releasetools/extras.txt is used.
If you want to customise it, define this in your BoardConfig:

TARGET_OTA_EXTRAS_FILE := device/VENDORNAME/DEVICENAME/extras.txt

Change-Id: I3d86914ccf262fc02b37d9789e21c1ec0d837701
---
 core/Makefile                            |  7 +++++++
 tools/releasetools/extras.txt            |  7 +++++++
 tools/releasetools/ota_from_target_files | 16 ++++++++++++++--
 3 files changed, 28 insertions(+), 2 deletions(-)
 create mode 100644 tools/releasetools/extras.txt

diff --git a/core/Makefile b/core/Makefile
index b6086b5..722d0e2 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1435,6 +1435,12 @@ else
     $(INTERNAL_OTA_PACKAGE_TARGET): override_device := $(TARGET_OTA_ASSERT_DEVICE)
 endif
 
+ifneq ($(TARGET_OTA_EXTRAS_FILE),)
+    $(INTERNAL_OTA_PACKAGE_TARGET): extras_file := $(TARGET_OTA_EXTRAS_FILE)
+else
+    $(INTERNAL_OTA_PACKAGE_TARGET): extras_file := build/tools/releasetools/extras.txt
+endif
+
 ifneq ($(TARGET_UNIFIED_DEVICE),)
     $(INTERNAL_OTA_PACKAGE_TARGET): override_prop := --override_prop=true
 endif
@@ -1449,6 +1455,7 @@ $(INTERNAL_OTA_PACKAGE_TARGET): $(BUILT_TARGET_FILES_PACKAGE) $(DISTTOOLS)
 	   -k $(KEY_CERT_PAIR) \
 	   --backup=$(backuptool) \
 	   --override_device=$(override_device) $(override_prop) \
+	   --extras_file=$(extras_file) \
 	   $(BUILT_TARGET_FILES_PACKAGE) $@
 
 .PHONY: otapackage bacon
diff --git a/tools/releasetools/ota_from_target_files b/tools/releasetools/ota_from_target_files
index f662f96..efa7f18 100755
--- a/tools/releasetools/ota_from_target_files
+++ b/tools/releasetools/ota_from_target_files
@@ -59,6 +59,9 @@ Usage:  ota_from_target_files [flags] input_target_files output_ota_package
   --override_device <device>
       Override device-specific asserts. Can be a comma-separated list.
 
+  --extras_file <file>
+      Extras file
+
   --override_prop <boolean>
       Override build.prop items with custom vendor init.
       Enabled when TARGET_UNIFIED_DEVICE is defined in BoardConfig
@@ -101,6 +104,7 @@ OPTIONS.aslr_mode = True
 OPTIONS.worker_threads = 3
 OPTIONS.backuptool = False
 OPTIONS.override_device = 'auto'
+OPTIONS.extras_file = 'build/tools/releasetools/extras.txt'
 OPTIONS.override_prop = False
 
 def MostPopularKey(d, default):
@@ -319,6 +323,7 @@ def CopySystemFiles(input_zip, output_zip=None,
   """
 
   symlinks = []
+  extras_list = [line.strip() for line in open(OPTIONS.extras_file)]
 
   for info in input_zip.infolist():
     if info.filename.startswith("SYSTEM/"):
@@ -328,7 +333,10 @@ def CopySystemFiles(input_zip, output_zip=None,
                          "/system/" + basefilename))
       else:
         info2 = copy.copy(info)
-        fn = info2.filename = "system/" + basefilename
+        if basefilename in extras_list:
+          fn = info2.filename = "extras/" + basefilename
+        else:
+          fn = info2.filename = "system/" + basefilename
         if substitute and fn in substitute and substitute[fn] is None:
           continue
         if output_zip is not None:
@@ -866,6 +874,8 @@ def main(argv):
       OPTIONS.backuptool = bool(a.lower() == 'true')
     elif o in ("--override_device"):
       OPTIONS.override_device = a
+    elif o in ("--extras_file"):
+      OPTIONS.extras_file = str(a)
     elif o in ("--override_prop"):
       OPTIONS.override_prop = bool(a.lower() == 'true')
     else:
@@ -884,6 +894,7 @@ def main(argv):
                                               "aslr_mode=",
                                               "backup=",
                                               "override_device=",
+                                              "extras_file=",
                                               "override_prop="],
                              extra_option_handler=option_handler)

-- 
1.8.5.2

